src/Microsoft.NetCore.Analyzers/Core/LeapYear/DoNotIncrementYearValueInDateConstructor.cs

```diff
@@ -0,0 +1,123 @@
+
+using System.Collections.Immutable;
+using System.Globalization;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.NetCore.Analyzers.LeapYear
+{
+    /// <summary>
+    /// CA2260: Do not increment or decrement year parameter in DateTime constructor.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class DoNotIncrementYearValueInDateConstructor : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA2260";
+
+        private static readonly LocalizableString _localizableTitle = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.DoNotIncrementYearInDateTimeConstructorTitle), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString _localizableMessage = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.DoNotIncrementYearInDateTimeConstructorMessage), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            RuleId,
+            _localizableTitle,
+            _localizableMessage,
+            DiagnosticCategory.Usage,
+            DiagnosticHelpers.DefaultDiagnosticSeverity,
+            isEnabledByDefault: false,
+            helpLinkUri: null);
+
+        private static readonly LocalizableString _localizableIdentifierTitle = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.DoNotIncrementYearInDateTimeIdentifierConstructorTitle), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString _localizableIdentifierMessage = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.DoNotIncrementYearInDateTimeIdentifierConstructorMessage), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+
+        internal static DiagnosticDescriptor IdentifierRule = new DiagnosticDescriptor(
+            RuleId,
+            _localizableIdentifierTitle,
+            _localizableIdentifierMessage,
+            DiagnosticCategory.Usage,
+            DiagnosticHelpers.DefaultDiagnosticSeverity,
+            isEnabledByDefault: false,
+            helpLinkUri: null);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule, IdentifierRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze);
+            context.EnableConcurrentExecution();
+            context.RegisterSemanticModelAction(semanticModelAnalysisContext =>
```

Comment:
Please don't register this action - there are performance issues associated with it. I think you need to move to using IOperation API and RegisterOperationAction for all semantic analysis of executable code.